pkg/operator/apiserver/audit/legacy.go

```diff
@@ -0,0 +1,89 @@
+package audit
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	assets "github.com/openshift/library-go/pkg/operator/apiserver/audit/bindata"
+	libgoapiserver "github.com/openshift/library-go/pkg/operator/configobserver/apiserver"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+)
+
+// GetAuditPolicies returns a config map that holds the audit policies for the target namespaces and name.
+// Note: the returned ConfigMap has Kind and APIVersion not set. This is responsibility of the caller
+//       when serializing it.
+func GetAuditPolicies(targetName, targetNamespace string) (*corev1.ConfigMap, error) {
+	cm := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: targetNamespace,
+			Name:      targetName,
+		},
+		Data: map[string]string{},
+	}
+
+	for _, profile := range []configv1.AuditProfileType{configv1.AuditProfileDefaultType, configv1.WriteRequestBodiesAuditProfileType, configv1.AllRequestBodiesAuditProfileType} {
+		policy, err := GetAuditPolicy(configv1.Audit{Profile: profile})
+		if err != nil {
+			return nil, err
+		}
+
+		policy.Kind = "Policy"
+		policy.APIVersion = auditv1.SchemeGroupVersion.String()
+
+		var buf bytes.Buffer
+		if err := auditYamlSerializer.Encode(policy, &buf); err != nil {
+			return nil, err
+		}
+
+		cm.Data[fmt.Sprintf("%s.yaml", strings.ToLower(string(profile)))] = buf.String()
+	}
+
+	return &cm, nil
+}
+
+// WithAuditPolicies is meant to wrap a standard Asset function usually provided by an operator.
+// It delegates to GetAuditPolicies when the filename matches the predicate for retrieving a audit policy config map for target namespace and name.
+func WithAuditPolicies(targetName string, targetNamespace string, assetDelegateFunc resourceapply.AssetFunc) resourceapply.AssetFunc {
+	return func(file string) ([]byte, error) {
+		if file == "audit-policies-cm.yaml" {
```

Comment:
nit:

```
if file != "audit-policies-cm.yaml" {
  return assetDelegateFunc(file)
}
```

(less nesting makes code more readable ;-)

Reply:
that's the preexisting, kind of, and will die (see move to legacy.go). Not worth to make prettier.